src/Pact/Typechecker.hs

```diff
@@ -1014,6 +1022,11 @@ toAST TBinding {..} = do
     t' <- mangleType aid <$> traverse toUserType t
     an <- trackNode t' aid
     v' <- toAST v
+    case (v,v') of
+      (TLam {},App _ fun _) -> do
+        debug $ "toAST: Tracking bound lambda " ++ show aid ++ " -> " ++ unpack (funName fun)
+        tcBoundLambdas %= M.insert aid fun
```

Comment:
```suggestion
        tcBoundLambdas.at aid ?= fun
```
This better matches your use of `preuse` with `ix` earlier. I find it better to use the `Data.Map` interface, or the Lens interface, but not mix them.